Scripts/General/generic_hex_grid.gd

```diff
@@ -60,6 +69,14 @@ func set_hex(coord : Vector2i, value : Variant) -> void:
 	hexes[coord.x][coord.y] = value
 
 
+func find(hex) -> Vector2i:
+	for x in width:
+		for y in height:
+			if hexes[x][y] == hex:
+				return Vector2i(x, y)
+	return Vector2i(-1, -1)
```

Comment:
having Vector2i(-1, -1) as returned magic value for not found is less than ideal
at least add
`const COORD_TILE_NOT_FOUND = Vector2i(-1, -1)`

also it would be nice to have a version with predicate function
as == on objects is often not really what you want

Reply:
also consider find variant
that returns an Array of hexes

it is better for scenarios with multiple or 0 matches

Reply:
Version with predicate is good idea -- can be added later, but now i need only the version with object. The version with many results is also good, can be added.

About these negative values -- I think it's obvious that (-1, -1) is bad coord, but ok, i can add this const. Pity that Godot does not have optional types.

Reply:
actually its worse than i thought
-1,-1 is a perfectly fine coordinate
there is nothing about coordinates that prevents them from being negative...
we only chose to put top left on 0,0

this is a king of weird hardcode that creates really bad bugs when its changed
i definitely want it fixed

Reply:
```
const MIN_INT = -9223372036854775808
const COORD_TILE_NOT_FOUND = Vector2i(MIN_INT , MIN_INT )
```